src/components/ReviewVote.jsx

```diff
@@ -0,0 +1,122 @@
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { patchReview } from "../utils/api";
+
+export default function ReviewVote({ review }) {
+    const [votes, Setvotes] = useState(0)
+    const [upVote, setUpVote] = useState(false)
+    const [resetUpVote, setResetUpVote] = useState(false)
+    const [downVote, setDownVote] = useState(false)
+    const [resetDownVote, setResetDownVote] = useState(false)
+    const [err, setErr] = useState(null)
+    const { review_id } = useParams()
+
+    const incUpVote = () => {
+        setErr(null)
+        Setvotes((curr) => {
+            return curr + 1
+        })
+        patchReview(review_id, 1).catch((error) => {
+            setErr(error)
+        })
+        setUpVote(true)
+        setResetUpVote(true)
+    }
+
+    const descUpVote = () => {
+        setErr(null)
+        Setvotes((curr) => {
+            return curr -1
+        })
+        patchReview(review_id, -1).catch((error) => {
+            setErr(error)
+        })
+        setUpVote(false)
+        setResetUpVote(null)
+    }
+
+    const incDownVote = () => {
+        setErr(null)
+        Setvotes((curr) => {
+            return curr -1
+        })
+        patchReview(review_id, -1).catch((error) => {
+            setErr(error)
+        })
+        setDownVote(true)
+        setResetDownVote(true)
+    }
+
+    const descDownVote = () => {
+        setErr(null)
+        Setvotes((curr) => {
+            return curr + 1
+        })
+        patchReview(review_id, 1).catch((error) => {
+            setErr(error)
+        })
+        setDownVote(false)
+        setResetDownVote(null)
+    }
+
```

Comment:
There is quite a lot of similar code within these voting functions. Getting the functionality working is definitely the first priority but now it is working you may want to have a look at how you could refactor to make this code more DRY. Probably not a priority for this week but definitely something to think about